SCClassLibrary/Common/Core/Document.sc

```diff
@@ -3,6 +3,33 @@ Document {
 	classvar <globalKeyDownAction, <globalKeyUpAction, <>initAction;
 	classvar <>autoRun = true;
 	classvar <asyncActions;
+
+	classvar <linkedIDE;
+	/* the IDE that will instantiate documents. It is that IDE's responsibility to implement
+	methods that may be invoked on it from the Document class:
+	  getQUuid
+	  newDocument
+	  setCurrentDocumentByQUuid
+	  open
+	  close
+	  setDocumentKeyDownEnabled
+	  setDocumentKeyUpEnabled
+	  setDocumentMouseDownEnabled
+	  setDocumentMouseUpEnabled
+	  setDocumentTextChangedEnabled
+	  setDocumentGlobalKeyDownEnabled
+	  setDocumentGlobalKeyUpEnabled
+	  setDocumentTitle
+	  setSelectionByQUuid
+	  setEditablebyQUuid
+	  setPromptsToSavebyQUuid
+	  removeDocUndoByQUuid
+
+	TODO: Add code to actually link `linkedIDE`
+	TODO: Replace all references to ScIDE with linkedIDE
+	*/
+
```

Comment:
Yes, I'd be totally open to that! Am I correct that in thinking that the advantage there is leaving the possibility open for alternate editors to link into that functionality, at the slight cost of additional work for this PR? I suppose Document should not really be in charge of choosing its linked IDE...